### PR TITLE
Add the `StackMainContent` component

### PR DESCRIPTION
**Pull request by jamesricky**
depends on https://github.com/vivid-planet/comet/pull/2701

## Description

This version of `MainContent` only adds content spacing and height when it's the last visible `StackSwitch`.

Using `StackMainContent` instead of `MainContent` prevents unintended or duplicate spacings in cases where multiple `MainContent` components are used inside nested `StackSwitch` components.

## Example of handmade "Add Product Variant" form in demo

In this case, the form, using `MainContent` for content spacing (inside a `StackSwitch` and `StackPage`) is nested inside `Tabs` which are also wrapped with `MainContent` for spacing around the `Tabs`. 

| Before   | After   |
| -------- | ------- |
| Add Variant Form Previously | Add Variant Form Now |

## Changeset

-   [x] I have verified if my change requires a changeset

## Related tasks and documents

https://vivid-planet.atlassian.net/browse/COM-1245

## Open TODOs/questions

-   [x] Merge parent PR https://github.com/vivid-planet/comet/pull/2701

### CODE REVIEW

File: packages/admin/admin/src/stack/useIsLastVisibleSwitch.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import { useStackApi } from "./Api";
+import { useStackSwitchApi } from "./Switch";
+
+export const useIsLastVisibleSwitch = () => {
```
Comment (johnnyomair) [inline packages/admin/admin/src/stack/useIsLastVisibleSwitch.ts:4]: Naming: Wouldn't useIsActiveStackSwitch be better?

File: packages/admin/admin/src/stack/useIsLastVisibleSwitch.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    const stackApi = useStackApi();
+    const stackSwitchApi = useStackSwitchApi();
+
+    if (stackApi && stackSwitchApi) {
+        const currentSwitchIndex = stackSwitchApi.id ? stackApi.switches.findIndex((i) => i.id === stackSwitchApi.id) : -1;
+        const nextSwitchShowsInitialPage = stackApi.switches[currentSwitchIndex + 1] && stackApi.switches[currentSwitchIndex + 1].isInitialPageActive;
+        return currentSwitchIndex === stackApi.switches.length - (nextSwitchShowsInitialPage ? 2 : 1);
+    }
```
Comment (johnnyomair) [inline packages/admin/admin/src/stack/useIsLastVisibleSwitch.ts:12]: We also have this in RouterTabs: https://github.com/vivid-planet/comet/blob/main/packages/admin/admin/src/tabs/RouterTabs.tsx#L116-L122

Could we use it here as well?

**Review by johnnyomair — COMMENTED** [review]
Do we need an additional component? Why not added the behavior to MainContent?

File: packages/admin/admin/src/stack/useIsLastVisibleSwitch.ts (hunk 1 above)
Comment (jamesricky) [reply packages/admin/admin/src/stack/useIsLastVisibleSwitch.ts:4]: Yes 😄

File: packages/admin/admin/src/stack/useIsLastVisibleSwitch.ts (hunk 2 above)
Comment (jamesricky) [reply packages/admin/admin/src/stack/useIsLastVisibleSwitch.ts:12]: I think so, I'll use the hook here as well 👍🏼

**Comment by johnnyomair** [comment]
> But wouldn't this be a breaking change? 🤔

IMO we could consider this a bug fix 😁

**Comment by jamesricky** [comment]
> Why not added the behavior to MainContent?

@johnnyomair after some consideration, I think there actually are valid use-cases for `MainContent` without the stack-hiding behavior. 

E.g., if you nest stack-pages in which there is no content that should be outside of the content-spacing (like a `Toolbar`) the outer content-spacing might still be required and would now be unintentionally removed. This would require additional nested instances of `MainContent`. 

So I think it's best to keep them as separate `MainContent` and `StackMainContent` components, similar to how we have separate `Toolbar` and `StackToolbar` components for example.

**Review by johnnyomair — APPROVED** [review]

**Pull request merged**